DatadogInternal/Sources/NetworkInstrumentation/URLSession/DatadogURLSessionDelegate.swift

```diff
@@ -0,0 +1,93 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// The `URLSession` delegate object which enables network requests instrumentation. **It must be
+/// used together with** `DatadogRUM` or `DatadogTrace`.
+///
+/// All requests made with the `URLSession` instrumented with this delegate will be intercepted by the SDK.
+@objc
+open class DatadogURLSessionDelegate: NSObject, URLSessionDataDelegate {
```

Comment:
Almost identical to the v1 `DDURLSessionDelegate `, it will redirect tasks to the `NetworkInstrumentationFeature` in the provided `core`.